### PR TITLE
Upgrade react-hotkeys and reactstrap

### PR DESCRIPTION
**Pull request by armenzg**
The older packages were triggering warnings about upgrading to React 17.

```
    console.warn node_modules/react-dom/cjs/react-dom.development.js:12357
      Warning: componentWillMount has been renamed, and is not recommended for use. See https://fb.me/react-unsafe-component-lifecycles for details.

      * Move code with side effects to componentDidMount, and set initial state in the constructor.
      * Rename componentWillMount to UNSAFE_componentWillMount to suppress this warning in non-strict mode. In React 17.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.

      Please update the following components: HotKeys
    console.warn node_modules/react-dom/cjs/react-dom.development.js:12357
      Warning: componentWillReceiveProps has been renamed, and is not recommended for use. See https://fb.me/react-unsafe-component-lifecycles for details.

      * Move data fetching code or side effects to componentDidUpdate.
      * If you're updating state whenever props change, refactor your code to use memoization techniques or move it to static getDerivedStateFromProps. Learn more at: https://fb.me/react-derived-state
      * Rename componentWillReceiveProps to UNSAFE_componentWillReceiveProps to suppress this warning in non-strict mode. In React 17.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.

      Please update the following components: Modal
    console.warn node_modules/react-dom/cjs/react-dom.development.js:12357
      Warning: componentWillUpdate has been renamed, and is not recommended for use. See https://fb.me/react-unsafe-component-lifecycles for details.

      * Move data fetching code or side effects to componentDidUpdate.
      * Rename componentWillUpdate to UNSAFE_componentWillUpdate to suppress this warning in non-strict mode. In React 17.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.

      Please update the following components: Modal
```

### CODE REVIEW

**Comment by armenzg** [comment]
@sarah-clements This PR removes the "yellow" warnings from the JS tests.

File: package.json [hunk 1]
```diff
@@ -59,7 +59,7 @@
     "react-split-pane": "0.1.89",
     "react-table": "6.10.3",
     "react-tabs": "3.0.0",
-    "reactstrap": "7.1.0",
+    "reactstrap": "8.2.0",
```
Comment (sarah-clements) [inline package.json]: We need to hold off on upgrading reactstrap until this bug is resolved: https://github.com/reactstrap/reactstrap/issues/1472

**Comment by armenzg** [comment]
Hi @sarah-clements , I've tested versions 8.0.0, 8.2.0 and 8.3.1 version (as recommended on the issue you linked) and it is working for all three:
image

Is there a different page I should be testing the tooltip working?

**Comment by sarah-clements** [comment]
> Hi @sarah-clements , I've tested versions 8.0.0, 8.2.0 and 8.3.1 version (as recommended on the issue you linked) and it is working for all three:
> image
> 
> Is there a different page I should be testing the tooltip working?

That tooltip does not use the reactstrap Tooltip component. You can find specific reactstrap tooltips on the compare view page. Basically, search in the codebase for wherever you see the `SimpleTooltip` component. I've already tested this though and the tooltips aren't rendering on the compare view and is rendering strangely on the alerts view.

File: ui/shared/SimpleTooltip.jsx [hunk 2]
```diff
@@ -25,7 +25,6 @@ export default class SimpleTooltip extends React.Component {
         <UncontrolledTooltip
           placement={placement}
           target={this.tooltipRef}
-          className="tooltip"
```
Comment (sarah-clements) [inline ui/shared/SimpleTooltip.jsx]: So now the Tooltip will work with this version! I think this was accidentally left in (bootstrap class) when this component was created.

**Comment by sarah-clements** [comment]
The maintainer has been helpful in helping me troubleshoot the issue, so this should be good to go once the yarn.lock file conflict is resolved.

**Comment by armenzg** [comment]
I have fixed the lock and force pushed.

**Comment by sarah-clements** [comment]
Actually, when you force pushed you didn't pick up my change to the SimpleTooltip file

**Comment by armenzg** [comment]
> Actually, when you force pushed you didn't pick up my change to the SimpleTooltip file

My apologies. I had not noticed. I only saw the comment; I didn't realize there was a commit. Thank you!

**Pull request merged**